pkg/controller/certificates/sync.go

```diff
@@ -104,6 +107,29 @@ func (c *Controller) Sync(ctx context.Context, crt *v1alpha1.Certificate) (err e
 		return nil
 	}
 
+	// check that certificate secret name is unique
+	namespaceCrts := c.certificateLister.Certificates(crtCopy.Namespace)
+	otherCrts, err := namespaceCrts.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	hasDuplicate := false
+	for _, otherCrt := range otherCrts {
+		hasDuplicate = hasDuplicate ||
```

Comment:
if we've found a duplicate, maybe it would be better to just `break` here to avoid looking over every other certificate